lib/new_relic/agent/instrumentation/redis/instrumentation.rb

```diff
@@ -52,21 +59,35 @@ def with_tracing(operation, statement: nil, database: nil)
     end
 
     def _nr_hostname
-      _nr_client.path ? Constants::LOCALHOST : _nr_client.host
+      _nr_redis_client_config.path ? Constants::LOCALHOST : _nr_redis_client_config.host
     rescue => e
-      NewRelic::Agent.logger.debug("Failed to retrieve Redis host: #{e}")
+      NewRelic::Agent.logger.debug("Failed to retrieve Redis host: #{e.class} - #{e.message}")
       Constants::UNKNOWN
     end
 
     def _nr_port_path_or_id
-      _nr_client.path || _nr_client.port
+      _nr_redis_client_config.path || _nr_redis_client_config.port
     rescue => e
-      NewRelic::Agent.logger.debug("Failed to retrieve Redis port_path_or_id: #{e}")
+      NewRelic::Agent.logger.debug("Failed to retrieve Redis port_path_or_id: #{e.class} - #{e.message}")
       Constants::UNKNOWN
     end
 
-    def _nr_client
-      @nr_client ||= self.is_a?(::Redis::Client) ? self : client.config
+    def _nr_redis_client_config
+      @nr_config ||= begin
```

Comment:
The `begin` and `end` here create a block of code for the memoization with `@nr_config`. That instance variable gets set equal to the result of the entire block.

Alternatively, we could use 2 separate methods, like so to call a method instead of an anonymous block.

```ruby
def important_value
  @important_value ||= initialize_important_value
end

def initialize_important_value
  # multiple line logic goes here
end
```

Reply:
Thanks for explaining!